mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java

```diff
@@ -63,12 +74,20 @@
 
     String location = job.get(InputFormatConfig.TABLE_LOCATION);
     return Arrays.stream(super.getSplits(job, numSplits))
-                 .map(split -> new HiveIcebergSplit((IcebergSplit) split, location))
-                 .toArray(InputSplit[]::new);
+        .map(split -> new HiveIcebergSplit((IcebergSplit) split, location))
+        .toArray(InputSplit[]::new);
   }
 
   @Override
   public boolean shouldSkipCombine(Path path, Configuration conf) {
     return true;
   }
+
+  private SearchArgument constructSearchArgument(JobConf job, ExprNodeGenericFuncDesc exprNodeDesc) {
+    SearchArgument searchArgument = CONSTRUCT_SARG_HIVE_2.invoke(job, exprNodeDesc);
+    if (searchArgument == null) {
```

Comment:
good observation, I checked if `ConvertAstToSearchArg` can return null, and couldnt find a code path where it might. 

But even if the null somehow inexplicably shows up, Hive 2 doesn't have `SearchArgumentFactory` with `create`, so the null will propagate.